rpcserver.go

```diff
@@ -1120,6 +1121,52 @@ func (r *rpcServer) ListPeers(ctx context.Context,
 	return resp, nil
 }
 
+// ChangePassword changes the password of the currently loaded encrypted wallet,
+// so it must be unlocked first.
+func (r *rpcServer) ChangePassword(ctx context.Context,
+	in *lnrpc.ChangePasswordRequest) (*lnrpc.ChangePasswordResponse, error) {
+
+	// Check macaroon to see if this is allowed.
+	if r.authSvc != nil {
+		if err := macaroons.ValidateMacaroon(ctx, "changepassword",
+			r.authSvc); err != nil {
+			return nil, err
+		}
+	}
+
+	privateWalletPw := in.CurrentPassword
+	publicWalletPw := in.CurrentPassword
+
+	// Check if the current password provided is blank or is the default
+	// wallet password. This most likely means the wallet came from a
+	// --noencryptwallet state. In that case, we set the defaults.
+	if len(in.CurrentPassword) == 0 ||
+		bytes.Compare(in.CurrentPassword, defaultPrivateWalletPw) == 0 {
+		privateWalletPw = defaultPrivateWalletPw
+		publicWalletPw = defaultPublicWalletPw
+	}
+
+	// Require the new password to have a length of at least 8 characters.
+	if len(in.NewPassword) < 8 {
+		return nil, fmt.Errorf("password must have at least " +
+			"8 characters")
+	}
+
+	// Grab the loaded wallet from the server and change its private and
+	// public passwords to the new password provided.
+	w := r.server.cc.wallet
+	err := w.ChangePrivatePassphrase(privateWalletPw, in.NewPassword)
+	if err != nil {
+		return nil, err
+	}
+	err = w.ChangePublicPassphrase(publicWalletPw, in.NewPassword)
```

Comment:
This might be a bit dangerous if changing the private passphrase succeeds, but not the public passphrase. Not sure if there's an easy way to make it atomic though 🤔

Reply:
Fixed.